### PR TITLE
Consider waiting and pending compaction tasks as well as running tasks in DruidCoordinatorSegmentCompactor

### PR DESCRIPTION
**Pull request by jihoonson**
`DruidCoordinatorSegmentCompactor` currently computes the available number of task slots for compaction based on the number of running compaction tasks, i.e., total number of task slots - number of running compaction tasks. This PR is to consider pending and waiting compaction tasks as well to compute more accurate available compaction task slots.

This change is [Reviewable](https://reviewable.io/reviews/druid-io/druid/5704)

### CODE REVIEW

**Comment by b-slim** [comment]
@jihoonson can you please provide more context like what this is improving? Thanks

File: server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentCompactor.java [hunk 1]
```diff
@@ -73,32 +76,24 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
...
-        final int numAvailableCompactionTaskSlots = numRunningCompactTasks > 0 ?
-                                                    compactionTaskCapacity - numRunningCompactTasks :
+        final int numAvailableCompactionTaskSlots = numNonCompleteCompactionTasks > 0 ?
+                                                    compactionTaskCapacity - numNonCompleteCompactionTasks :
                                                     // compactionTaskCapacity might be 0 if totalWorkerCapacity is low.
                                                     // This guarantees that at least one slot is available if
                                                     // compaction is enabled and numRunningCompactTasks is 0.
                                                     Math.max(1, compactionTaskCapacity);
-        LOG.info("Running tasks [%d/%d]", numRunningCompactTasks, compactionTaskCapacity);
+        LOG.info("Running tasks [%d/%d]", numNonCompleteCompactionTasks, compactionTaskCapacity);
```
Comment (b-slim) [inline server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentCompactor.java]: can we make this more readable something like ` Total running and waiting tasks [], Maximum compaction queue capacity [] `. Thanks

**Comment by b-slim** [comment]
👍  left minor comment.

File: server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentCompactor.java (hunk 1 above)
Comment (jihoonson) [reply server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentCompactor.java]: Good point. Fixed.

**Comment by jihoonson** [comment]
@b-slim do you have more comments?

**Comment by b-slim** [comment]
nop was waiting if someone else want to review it, i will merge it by end of day if no one look at this.

**Comment by jihoonson** [comment]
@b-slim thanks.

**Pull request merged**